code/game/objects/effects/decals/Cleanable/fuel.dm

```diff
@@ -1,60 +1,61 @@
-				if(origin.CanPass(null, target) && target.CanPass(null, origin))
-					if(!locate(/obj/effect/decal/cleanable/liquid_fuel) in target)
-						new/obj/effect/decal/cleanable/liquid_fuel(target, amount * 0.25, 1)
-						amount *= 0.75
-
-	flamethrower_fuel
-		icon_state = "mustard"
-		anchored = 0
-		New(newLoc, amt = 1, d = 0)
-			setDir(d) //Setting this direction means you won't get torched by your own flamethrower.
-			. = ..()
-
-		Spread()
-			//The spread for flamethrower fuel is much more precise, to create a wide fire pattern.
-			if(amount < 0.1) return
-			var/turf/S = loc
-			if(!istype(S)) return
-
-			for(var/d in list(turn(dir,90),turn(dir,-90), dir))
-				var/turf/O = get_step(S,d)
-				if(locate(/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel) in O)
-					continue
-				if(O.CanPass(null, S) && S.CanPass(null, O))
-					new/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel(O, amount * 0.25, 1)
+/obj/effect/decal/cleanable/liquid_fuel/Initialize(turf/newLoc, amt = 1, nologs = 0)
+	if(!nologs)
+		log_game("[amt] units of liquid fuel have spilled in [AREACOORD(newLoc.loc)].")
+		message_admins("[amt] units of liquid fuel have spilled in [ADMIN_VERBOSEJMP(newLoc.loc)].")
+	amount = amt
+
+	//Be absorbed by any other liquid fuel in the tile.
+	for(var/obj/effect/decal/cleanable/liquid_fuel/other in newLoc)
+		if(other != src)
+			other.amount += src.amount
+			spawn other.Spread()
+			qdel(src)
+
+	Spread()
+	. = ..()
+
+/obj/effect/decal/cleanable/liquid_fuel/proc/Spread()
+	//Allows liquid fuels to sometimes flow into other tiles.
+	if(amount < 5.0) return
+	var/turf/S = loc
+	if(!istype(S)) return
+	for(var/d in cardinal)
+		if(rand(25))
+			var/turf/target = get_step(src,d)
+			var/turf/origin = get_turf(src)
+			if(origin.CanPass(null, target) && target.CanPass(null, origin))
+				if(!locate(/obj/effect/decal/cleanable/liquid_fuel) in target)
+					new/obj/effect/decal/cleanable/liquid_fuel(target, amount * 0.25, 1)
+					amount *= 0.75
+
+/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel
+	icon_state = "mustard"
+	anchored = 0
+
+/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel/Initialize(newLoc, amt = 1, d = 0)
+	setDir(d) //Setting this direction means you won't get torched by your own flamethrower.
+	. = ..()
+
+/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel/Spread()
+	//The spread for flamethrower fuel is much more precise, to create a wide fire pattern.
+	if(amount < 0.1) return
+	var/turf/S = loc
+	if(!istype(S)) return
+
+	for(var/d in list(turn(dir,90),turn(dir,-90), dir))
+		var/turf/O = get_step(S,d)
+		if(locate(/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel) in O)
+			continue
+		if(O.CanPass(null, S) && S.CanPass(null, O))
+			new/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel(O, amount * 0.25, 1)
 //					O.hotspot_expose((T20C * 2) + 380, 500) //Light flamethrower fuel on fire immediately.
 
-			amount *= 0.25
+	amount *= 0.25
```

Comment:
Intended indentation ?

Reply:
its in line with the for loop above it still.

Reply:
So, whether it succeed to spread or not, it loses 75% of it content ?

Probably was intended to be :
```suggestion
			amount *= 0.75
```

Reply:
Yes, if you use side-by-side diffs you'll see this line was the same indent as line 52/53 before and is still the same indent now

Reply:
It's reduced to 25% after spreading into 3 tiles (`list(turn(dir,90),turn(dir,-90), dir)`), 25% in each, for a total of 100%.
Works differently than the other fuel.

Reply:
Except it is reduced to this amount whether it succeed in spreading or not.
So any `/obj/effect/decal/cleanable/liquid_fuel/flamethrower_fuel` is instanciated with 25% of its amount even if it didn't spread (using against a wall, throwing fuel over fuel, etc).
Anyway, it's not critical, it can stay that way for now.